asdf/config.py

```diff
@@ -0,0 +1,81 @@
+from .util import NotSet
+from .providers import get_registered_converter_providers, create_converters
+from .converter import ConverterIndex
+
+
+__all__ = ["get_config", "configure"]
+
+
+class AsdfConfig:
+    def __init__(
+        self,
+        converter_providers=None,
+        converters=None,
+        converter_index=None,
+    ):
+        self._converter_providers = converter_providers
+        self._converters = converters
+        self._converter_index = converter_index
+
+    @property
+    def converter_providers(self):
+        if self._converter_providers is None:
+            self._converter_providers = get_registered_converter_providers()
+
+        return self._converter_providers
+
+    @property
+    def converters(self):
+        if self._converters is None:
+            self._converters = create_converters(self.converter_providers)
+
+        return self._converters
+
+    @property
+    def converter_index(self):
+        if self._converter_index is None:
+            self._converter_index = ConverterIndex(self.converters)
+
+        return self._converter_index
+
+
+_config = AsdfConfig()
+
+
+def get_config():
+    return _config
+
+
+class _ConfigContext:
+    def __init__(self, original_config):
```

Comment:
I have a preference for something like current_config or previous_config. Original makes me think it is the first.

Reply:
I like "previous", I'll change it.

Reply:
(the config module changed enough that this is no longer relevant)